interface/app/$libraryId/settings/library/locations/IndexerRuleEditor/RulesForm.tsx

```
@@ -0,0 +1,326 @@
+		formState: { errors }
+	} = form;
+
+	const { fields, append, remove } = useFieldArray({
+		control: form.control,
+		name: 'rules'
+	});
+
+	//this is used to update the input type based on rule 'type' selected
+	const memoWatchRules = useCallback(
+		(index: number) => {
+			return form.watch(`rules.${index}.type`) as InputKinds;
+		},
+		[form]
+	);
+
+	const validateInput = (
+		type: InputKinds,
+		value: string
+	): { value: any; message: string } | undefined => {
+		switch (type) {
+			case 'Extension': {
+				const extensionRegex = /^\.[^.\s]+$/;
+				return {
+					value: extensionRegex.test(value),
+					message: value ? 'Invalid extension' : 'Value required'
+				};
+			}
+			case 'Name': {
+				return {
+					value: value ? true : false,
+					message: 'Value required'
+				};
+			}
+			case 'Path': {
+				return {
+					value: value ? true : false,
+					message: 'Value required'
+				};
+			}
+			case 'Advanced': {
+				return {
+					value: value ? true : false,
+					message: 'Value required'
+				};
+			}
+			default:
+				return undefined;
+		}
+	};
+
+	const inputValidator = (
+		index: number,
+		watcher: InputKinds,
+		e: ChangeEvent<HTMLInputElement>
+	) => {
+		const isValid = validateInput(watcher, e.target.value);
+		if (!isValid?.value) {
+			form.setError(`rules.${index}.value`, {
+				message: isValid?.message
+			});
+		} else {
+			form.clearErrors(`rules.${index}.value`);
+		}
+	};
+
+	const addIndexerRules = useCallback(
+		(data: formType) => {
+			const formatData = {
+				kind: data.kind,
+				name: data.name,
+				dry_run: false,
+				parameters: data.rules.flatMap(({ type, value }) => {
+					switch (type) {
+						case 'Name':
+							return `**/${value}`;
+						case 'Extension':
+							// .tar should work for .tar.gz, .tar.bz2, etc.
+							return [`**/*${value}`, `**/*${value}.*`];
+						default:
+							return value;
+					}
+				})
+			} as IndexerRuleCreateArgs;
+			createIndexerRules.mutateAsync(formatData);
```

This promise should be either awaited or returned, as it is right now backend errors will be ignored by the frontend